.ci/magic-modules/generate-chef.sh

```diff
@@ -0,0 +1,44 @@
+#!/bin/bash
```

Comment:
All Puppet & Chef scripts can be the same aside from the folder and YAML files they source/output:

source: `<provider>.yaml`, e.g. `puppet.yaml`
output: `build/<provider>/<product`, e.g. `build/chef/compute`

Avoid massive copy/paste and favor reuse/generalize the scripts. This will save maintenance efforts in the long run.